common/StringVector.hpp

```diff
@@ -51,19 +52,35 @@ class StringVector
     }
 
     /// Tokenize delimited values until we hit new-line or the end.
+    /// Ignoring delimiters that are within quotation marks, unless ignoreQuotes is set to true
     static void tokenize(const char* data, const std::size_t size, const char delimiter,
-                         std::vector<StringToken>& tokens)
+                         std::vector<StringToken>& tokens, bool ignoreQuotes = true)
```

Comment:
I'm not a fan of reusing function with default parameters. They make maintenance harder later on.

I also worry about the cost of the new logic. These tokenerizers are heavily used for each and every command, and multiple times at that. Perhaps we should consider separating the cases where we expect quotes vs those that we never support quotes and have separate implementations?